### DISCUSSION TITLE
label_names() support for Prometheus match parameter

### DISCUSSION THREAD

**Discussion by callumj** [grafana/grafana#52947] [Data Sources]
The [underlying API](https://prometheus.io/docs/prometheus/latest/querying/api/#getting-label-names) that powers `label_names()` supports the provide a selector (`match`) to allows for scoping the label names lookup to a metric. Is this something Grafana could support as an optional argument to `label_names()`?

**Comment by gtk-grafana** [comment]
So this has been implemented and merged in 9.2.5 in https://github.com/grafana/grafana/pull/58087. If you set your Prometheus [type](https://grafana.com/docs/grafana/latest/datasources/prometheus/#configure-the-data-source) and version in your data source config UI or in the [provisioning file](https://grafana.com/docs/grafana/latest/datasources/prometheus/#configure-the-data-source), supported versions will get the new API endpoints, and unsupported (or empty) versions will get the old fallback behavior.

<img width="579" alt="image" src="https://github.com/grafana/grafana/assets/109082771/1d476b0f-eec1-4513-b599-6523aa78ee84">

Eventually we'd like to invert this behavior, and by default everyone gets new API endpoints, and only users that select older version will get compatibility API endpoints, but this isn't on the roadmap yet. If you think it should be (or have any other questions/comments) let us know!

**Comment by callumj** [comment]
@gtk-grafana What we're looking for is something like this: `label_names(metric='citadel_server_csr_count')` as an improvement to the variable function `label_names()` to allow for tighter scoping. Is that supported?

**Reply by gtk-grafana** [reply]
Ahhh, sorry I misunderstood earlier. I'll take a quick look and get an issue created for this!

**Comment by gtk-grafana** [comment]
A solution for this was [merged into 10.1.x](https://github.com/grafana/grafana/pull/70452). Thank you @callumj for the heads up!
